### PR TITLE
feat: support pull_request_target events

### PR DESCRIPTION
**Pull request by kadams54**
## Description

`pull_request` events don't have the necessary privileges to access private repos. Adding support for the `pull_request_target` event will allow private repos to also use commit lint.

## Related Issue

[If your pull request is related to an existing issue, reference it here using the format "Fixes #issue_number".]

## Type of Change

Please mark the appropriate option below to describe the type of change your pull request introduces:

-   [ ] Bug fix
-   [x] New feature
-   [ ] Enhancement
-   [ ] Documentation update
-   [ ] Refactor
-   [ ] Other (please specify)

## Checklist

-   [x] My pull request has a clear title and description.
-   [x] I have used [semantic commit messages](https://seesparkbox.com/foundry/semantic_commit_messages).
        Examples: `"fix: Fixed foobar bug"`, `"feat(accounts): Added foobar feature"`.
-   [x] I have added/updated the necessary documentation on `README.md`.
-   [x] I have added appropriate test cases (if applicable) to ensure the changes are functioning correctly.

## Additional Notes

[Add any additional notes or context that you think the reviewers should know about.]

By submitting this pull request, I confirm that I have read and complied with the contribution guidelines of this project.

### CODE REVIEW

**Comment by aj3sh** [comment]
Hi @kadams54, Thank you for contributing and adding a new feature in the commitlint. We will quickly review the code and the `pull_request_target` event details.
This is a must-have feature that we have missed.

**Review by aj3sh — DISMISSED** [review]
Looks Good.

File: github_actions/run.py [hunk 1]
```diff
@@ -176,6 +177,8 @@ def main() -> None:
         _handle_push_event(event)
     elif event.event_name == EVENT_PULL_REQUEST:
         _handle_pr_event(event)
+    elif event.event_name == EVENT_PULL_REQUEST_TARGET:
+        _handle_pr_event(event)
```
Comment (sugat009) [inline github_actions/run.py:181]:
```suggestion
    elif event.event_name in {EVENT_PULL_REQUEST, EVENT_PULL_REQUEST_TARGET}:
        _handle_pr_event(event)
```

**Review by sugat009 — COMMENTED** [review]
@kadams54 Thanks for your contribution. I've left a suggestion on the changes.

File: github_actions/run.py (hunk 1 above)
Comment (sugat009) [reply github_actions/run.py:181]: Or better yet, declare a constant with the events and use it in the if clause.
For eg:
```python
...
PULL_REQUEST_EVENTS = {EVENT_PULL_REQUEST, EVENT_PULL_REQUEST_TARGET}
...
elif event.event_name in PULL_REQUEST_EVENTS:
    _handle_pr_event(event)
...
```

**Review by sugat009 — APPROVED** [review]

**Review by aj3sh — APPROVED** [review]

**Pull request merged**